pkg/connection/connection.go

```diff
@@ -0,0 +1,420 @@
+package connection
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"database/sql/driver"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"os"
+	"os/user"
+	"runtime"
+	"strconv"
+
+	"github.com/exasol/exasol-driver-go/internal/config"
+	"github.com/exasol/exasol-driver-go/internal/utils"
+	"github.com/exasol/exasol-driver-go/internal/version"
+	"github.com/exasol/exasol-driver-go/pkg/errors"
+	"github.com/exasol/exasol-driver-go/pkg/logger"
+	"github.com/exasol/exasol-driver-go/pkg/proxy"
+	"github.com/exasol/exasol-driver-go/pkg/types"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/gorilla/websocket"
+)
+
+type Connection struct {
+	Config    *config.Config
+	websocket *websocket.Conn
+	Ctx       context.Context
+	IsClosed  bool
+}
+
+func (c *Connection) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
+	values, err := utils.NamedValuesToValues(args)
+	if err != nil {
+		return nil, err
+	}
+	return c.query(ctx, query, values)
+}
+
+func (c *Connection) ExecContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Result, error) {
+	values, err := utils.NamedValuesToValues(args)
+	if err != nil {
+		return nil, err
+	}
+	return c.exec(ctx, query, values)
+}
+
+func (c *Connection) Exec(query string, args []driver.Value) (driver.Result, error) {
+	return c.exec(context.Background(), query, args)
+}
+
+func (c *Connection) Query(query string, args []driver.Value) (driver.Rows, error) {
+	return c.query(context.Background(), query, args)
+}
+
+func (c *Connection) PrepareContext(ctx context.Context, query string) (driver.Stmt, error) {
+	if c.IsClosed {
+		logger.ErrorLogger.Print(errors.ErrClosed)
+		return nil, driver.ErrBadConn
+	}
+	response := &types.CreatePreparedStatementResponse{}
+	err := c.createPreparedStatement(ctx, query, response)
+	if err != nil {
+		return nil, err
+	}
+	return c.CreateStatement(response), nil
+}
+
+func (c *Connection) createPreparedStatement(ctx context.Context, query string, response *types.CreatePreparedStatementResponse) error {
```

Comment:
Can you remove the `response` parameter, create a new `CreatePreparedStatementResponse` in this method and return it? Then callers don't need to do that every time.